lib/hooks.js

```diff
@@ -5,50 +5,96 @@ const { logger } = require('./utils/logger');
 const debug = logger.debugContext('hooks');
 
 const hookTypes = {
-  beforeValidate: { params: 2 },
```

Comment:
We can also consider `Model.hooks.beforeValidate(() => {])` in addition to `Model.hooks.add('beforeValidate', () => {})`, thus adding these methods directly in `Hooks` class. Syntax is bit shorter

Then we can document each method properly.

Reply:
That'd kinda defeat the point of this tho, hooks don't know what they merely act as a promise event emitter.

Hooks that might be present in one context might not be in another.